DEVELOPMENT.md

```diff
@@ -0,0 +1,213 @@
+# Local Dev Installation
+
+This project has dependencies on Go >=1.21, protobuf-compiler and the corresponding Go plugins, and node >= 20.
+
+## Ubuntu
+
+Do not use apt to install any dependencies, the versions they install are all too old.
+Script below will curl latest versions and install them.
+
+```sh
+# Standard Go installation script
+curl -O https://dl.google.com/go/go1.21.1.linux-amd64.tar.gz
+sudo rm -rf /usr/local/go
+sudo tar -C /usr/local -xzf go1.21.1.linux-amd64.tar.gz
+echo 'export PATH=$PATH:/usr/local/go/bin' >> $HOME/.bashrc
+echo 'export GOPATH=$HOME/go' >> $HOME/.bashrc
+echo 'export PATH=$PATH:$GOPATH/bin' >> $HOME/.bashrc
+source $HOME/.bashrc
+
+cd cata
+
+# Install protobuf compiler and Go plugins
+sudo apt update && sudo apt upgrade
+sudo apt install protobuf-compiler
+go get -u -v google.golang.org/protobuf
+go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
+
+# Install node
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
+nvm install 20.13.1
+
+# Install the npm package dependencies using node
+npm install
+```
+
+## Docker
+
+Alternatively, install Docker and your workflow will look something like this:
+
+```sh
+git clone https://github.com/wowsims/cata.git
+cd cata
+
+# Build the docker image and install npm dependencies (only need to run these once).
+docker build --tag wowsims-cata .
+docker run --rm -v $(pwd):/cata wowsims-cata npm install
+
+# Now you can run the commands as shown in the Commands sections, preceding everything with, "docker run --rm -it -p 8080:8080 -v $(pwd):/cata wowsims-cata".
+# For convenience, set this as an environment variable:
+CATA_CMD="docker run --rm -it -p 8080:8080 -v $(pwd):/cata wowsims-cata"
+
+#For the watch commands assign this environment variable:
+CATA_WATCH_CMD="docker run --rm -it -p 8080:8080 -p 3333:3333 -p 5173:5173 -e WATCH=1 -v $(pwd):/cata wowsims-cata"
+
+# ... do some coding on the sim ...
+
+# Run tests
+$(echo $CATA_CMD) make test
+
+# ... do some coding on the UI ...
+
+# Host a local site
+$(echo $CATA_CMD) make host
+```
+
+## Windows
+
+If you want to develop on Windows, we recommend setting up a Ubuntu virtual machine (VM) or running Docker using [this guide](https://docs.docker.com/desktop/windows/wsl/ 'https://docs.docker.com/desktop/windows/wsl/') and then following the Ubuntu or Docker instructions, respectively.
+
+## Mac OS
+
+-   Docker is available in OS X as well, so in theory similar instructions should work for the Docker method
+-   You can also use the Ubuntu setup instructions as above to run natively, with a few modifications:
+    -   You may need a different Go installer if `go1.18.3.linux-amd64.tar.gz` is not compatible with your system's architecture; you can do the Go install manually from `https://go.dev/doc/install`.
```

Comment:
Looks like this still references an older Go version

Reply:
I would probably recommend that people download Go directly from the official Go website just to make sure that people get the right version for their architecture (thanks Apple)